### PR TITLE
Split live photo listener to extract trashbin specific code into its own listener

### LINKED ISSUE #43299: [Bug]: \OCA\Files\Listener\SyncLivePhotosListener requires files_trashbin to be enabled
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I updated NC Server 27.1.6 to 28.0.2.5 by updater.phar.
I had to deactivate several apps (e.g. fulltextsearch tesseract because of incompatibility to NC 28.x)
For a long time i also disabled the files_trashbin app - ok for NC 24 .... 27 but now with NC 28 i get errors in logfile

```
[no app in context] Fehler: Could not load event listener service OCA\Files\Listener\SyncLivePhotosListener: Could not resolve OCA\Files_Trashbin\Trash\ITrashManager! Class "OCA\Files_Trashbin\Trash\ITrashManager" does not exist. Make sure the class is auto-loadable by the Nextcloud server container
	von ? von -- um 01.02.2024, 16:10:05

```

### Steps to reproduce

1. disable files_trashbin app in an earlier NC version than 28 (e.g. 27.1.6)
2. update your NC e.g. script updater.phar to latest NC 28.0.2.5
3. check logfiles that a missing class is flooding your log with errors (see description)

### Expected behavior

The apps section is in users choice a list of optional features and functionalities and so it is not possible for the moment to cut of a "dependency" between the disabled "files_trashbin" app and a supression of messages concerning a missing class in nextcloud and / or the logfiles.

Please make the files_trashbin app again optional without having those errors in the logfile.

 

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.55.16",
            "MY.DOMAIN"
        ],
        "dns_pinning": true,
        "activity_expire_days": 14,
        "auth.bruteforce.protection.enabled": true,
        "blacklisted_files": [
            ".htaccess",
            "Thumbs.db",
            "thumbs.db"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "MY.DOMAIN",
        "overwriteconaddr": "^192\\.168\\.55\\.90$",
        "overwriteprotocol": "https",
        "forward_for_headers": [
            "X_FORWARDED_FOR",
            "HTTP_X_FORWARDED_FOR"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/MY.DOMAIN",
        "dbtype": "pgsql",
        "version": "28.0.2.5",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "logtimezone": "Europe\/Berlin",
        "default_phone_region": "DE",
        "logfile": "\/media\/cloud\/data\/nextcloud.log",
        "log_rotate_size": 10485760,
        "cron_log": true,
        "installed": true,
        "filesystem_check_changes": 1,
        "quota_include_external_storage": false,
        "skeletondirectory": "",
        "share_folder": "\/Freigaben",
        "knowledgebaseenabled": false,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "ldapUserCleanupInterval": 20,
        "bulkupload.enabled": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "mail_smtpmode": "sendmail",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtptimeout": 10,
        "mail_smtpauth": 0,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "integrity.check.disabled": false,
        "updater.release.channel": "stable",
        "enable_previews": true,
        "preview_Movie_path": "\/usr\/bin\/ffmpeg",
        "preview_ffmpeg_path": "\/usr\/bin\/ffmpeg",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\Movie",
            "OC\\Preview\\PDF",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\MSOffice2003",
            "OC\\Preview\\MSOfficeDoc",
            "OC\\Preview\\Image",
            "OC\\Preview\\Photoshop",
            "OC\\Preview\\SVG",
            "OC\\Preview\\Font",
            "OC\\Preview\\MKV",
            "OC\\Preview\\MP4",
            "OC\\Preview\\AVI",
            "OC\\Preview\\TIFF"
        ],
        "preview_max_x": 2048,
        "preview_max_y": 2048,
        "preview_max_memory": 4096,
        "preview_max_filesize_image": 256,
        "preview_max_scale_factor": 1,
        "trashbin_retention_obligation": "auto, 2",
        "loglevel": 3,
        "mail_sendmailmode": "smtp",
        "data-fingerprint": "FINGERPRINT",
        "onlyoffice": {
            "verify_peer_off": true,
            "jwt_header": "Authorization",
            "jwt_secret": "***REMOVED SENSITIVE VALUE***"
        },
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false,
        "defaultapp": "dashboard",
        "theme": "",
        "allow_local_remote_servers": true,
        "files_external_allow_create_new_local": true,
        "simpleSignUpLink.shown": false,
        "memories.exiftool": "\/var\/www\/nextcloud\/apps\/memories\/bin-ext\/exiftool-amd64-glibc",
        "memories.vod.path": "\/var\/www\/nextcloud\/apps\/memories\/bin-ext\/go-vod-amd64",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "memories.gis_type": 2,
        "session_keepalive": true,
        "memories.vod.disable": false,
        "memories.video_default_quality": "-2",
        "maintenance_window_start": 1
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - bookmarks: 13.1.3
  - bruteforcesettings: 2.8.0
  - calendar: 4.6.4
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contacts: 5.5.1
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_accesscontrol: 1.18.0
  - files_automatedtagging: 1.18.0
  - files_external: 1.20.0
  - files_fulltextsearch: 28.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - fulltextsearch: 28.0.0
  - fulltextsearch_elasticsearch: 28.0.0
  - gpxpod: 5.0.15
  - groupfolders: 16.0.3
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - memories: 6.2.2
  - nextcloud_announcements: 1.17.0
  - notes: 4.9.2
  - notifications: 2.16.0
  - notify_push: 0.6.9
  - oauth2: 1.16.3
  - onlyoffice: 9.0.0
  - password_policy: 1.18.0
  - photos: 2.4.0
  - previewgenerator: 5.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - support: 1.11.0
  - suspicious_login: 6.0.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - theming_customcss: 1.15.0
  - twofactor_backupcodes: 1.17.0
  - twofactor_totp: 10.0.0-beta.2
  - twofactor_webauthn: 1.3.2
  - updatenotification: 1.18.0
  - user_ldap: 1.19.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - encryption: 2.16.0 (installed 2.3.0)
  - files_fulltextsearch_tesseract: 27.0.0 (installed 27.0.0)
  - firstrunwizard: 2.17.0 (installed 2.3.0)
  - recommendations: 2.0.0 (installed 1.3.0)
  - survey_client: 1.16.0 (installed 1.2.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"NnW0qEbcJMCDGXXc7MJK","level":3,"time":"2024-02-01T16:10:05+01:00","remoteAddr":"","user":"--","app":"no app in context","method":"","url":"--","message":"Could not load event listener service OCA\\Files\\Listener\\SyncLivePhotosListener: Could not resolve OCA\\Files_Trashbin\\Trash\\ITrashManager! Class \"OCA\\Files_Trashbin\\Trash\\ITrashManager\" does not exist. Make sure the class is auto-loadable by the Nextcloud server container","userAgent":"--","version":"28.0.2.5","exception":{"Exception":"OCP\\AppFramework\\QueryException","Message":"Could not resolve OCA\\Files_Trashbin\\Trash\\ITrashManager! Class \"OCA\\Files_Trashbin\\Trash\\ITrashManager\" does not exist","Code":0,"Trace":[{"function":"OC\\AppFramework\\Utility\\{closure}","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":83,"function":"array_map"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":128,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":146,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":468,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/ServerContainer.php","line":155,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/ServiceEventListener.php","line":70,"function":"query","class":"OC\\ServerContainer","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":230,"function":"__invoke","class":"OC\\EventDispatcher\\ServiceEventListener","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":59,"function":"callListeners","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":94,"function":"dispatch","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":106,"function":"dispatch","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Cache/Cache.php","line":569,"function":"dispatchTyped","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Cache/Updater.php","line":164,"function":"remove","class":"OC\\Files\\Cache\\Cache","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":301,"function":"remove","class":"OC\\Files\\Cache\\Updater","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":1173,"function":"removeUpdate","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":702,"function":"basicOperation","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Storage.php","line":262,"function":"unlink","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Storage.php","line":935,"function":"deleteVersion","class":"OCA\\Files_Versions\\Storage","type":"::"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Command/Expire.php","line":51,"function":"expire","class":"OCA\\Files_Versions\\Storage","type":"::"},{"file":"/var/www/nextcloud/lib/private/Command/CommandJob.php","line":35,"function":"handle","class":"OCA\\Files_Versions\\Command\\Expire","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/Job.php","line":54,"function":"run","class":"OC\\Command\\CommandJob","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/QueuedJob.php","line":47,"function":"execute","class":"OC\\BackgroundJob\\Job","type":"->"},{"file":"/var/www/nextcloud/cron.php","line":152,"function":"execute","class":"OC\\BackgroundJob\\QueuedJob","type":"->"}],"File":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","Line":114,"Previous":{"Exception":"OC\\AppFramework\\Utility\\QueryNotFoundException","Message":"Could not resolve OCA\\Files_Trashbin\\Trash\\ITrashManager! Class \"OCA\\Files_Trashbin\\Trash\\ITrashManager\" does not exist","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":146,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/ServerContainer.php","line":173,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":443,"function":"query","class":"OC\\ServerContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":96,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->"},{"function":"OC\\AppFramework\\Utility\\{closure}","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":83,"function":"array_map"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":128,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":146,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":468,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/ServerContainer.php","line":155,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/ServiceEventListener.php","line":70,"function":"query","class":"OC\\ServerContainer","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":230,"function":"__invoke","class":"OC\\EventDispatcher\\ServiceEventListener","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":59,"function":"callListeners","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":94,"function":"dispatch","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":106,"function":"dispatch","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Cache/Cache.php","line":569,"function":"dispatchTyped","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Cache/Updater.php","line":164,"function":"remove","class":"OC\\Files\\Cache\\Cache","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":301,"function":"remove","class":"OC\\Files\\Cache\\Updater","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":1173,"function":"removeUpdate","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":702,"function":"basicOperation","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Storage.php","line":262,"function":"unlink","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Storage.php","line":935,"function":"deleteVersion","class":"OCA\\Files_Versions\\Storage","type":"::"},{"file":"/var/www/nextcloud/apps/files_versions/lib/Command/Expire.php","line":51,"function":"expire","class":"OCA\\Files_Versions\\Storage","type":"::"},{"file":"/var/www/nextcloud/lib/private/Command/CommandJob.php","line":35,"function":"handle","class":"OCA\\Files_Versions\\Command\\Expire","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/Job.php","line":54,"function":"run","class":"OC\\Command\\CommandJob","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/QueuedJob.php","line":47,"function":"execute","class":"OC\\BackgroundJob\\Job","type":"->"},{"file":"/var/www/nextcloud/cron.php","line":152,"function":"execute","class":"OC\\BackgroundJob\\QueuedJob","type":"->"}],"File":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","Line":135},"message":"Could not load event listener service OCA\\Files\\Listener\\SyncLivePhotosListener: Could not resolve OCA\\Files_Trashbin\\Trash\\ITrashManager! Class \"OCA\\Files_Trashbin\\Trash\\ITrashManager\" does not exist. Make sure the class is auto-loadable by the Nextcloud server container","exception":[],"CustomMessage":"Could not load event listener service OCA\\Files\\Listener\\SyncLivePhotosListener: Could not resolve OCA\\Files_Trashbin\\Trash\\ITrashManager! Class \"OCA\\Files_Trashbin\\Trash\\ITrashManager\" does not exist. Make sure the class is auto-loadable by the Nextcloud server container"},"id":"65bd1e19ef3f3"}
```

### Additional info

I activated the files_trashbin app for the moment to get calm the logs 

So you can see this app as activated in the app:list section for now but this is just my personal "workaround".

The problem is also discussed here:
https://help.nextcloud.com/t/queryexception-could-not-resolve-oca-files-trashbin-trash-itrashmanager/180560/2

### PR DESCRIPTION
**Pull request by artonge**
Fix https://github.com/nextcloud/server/issues/43299

## Done

- [x] Split live photo listener to extract trash specific code into its folder
- [x] Write e2e code to ensure nothing broke
- [x] Disable ajax cron to make e2e tests faster and more reliable
- [x] Update scroll strategy in e2e tests to ensure files row are not hided under the static top bar

## TODO

- [x] Backport to stable28

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: cypress/e2e/files/live_photos.cy.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		it("Shows both files when 'Show hidden files' is enabled", () => {
+			getRowForFileId(jpgFileId).should('have.length', 1).invoke('attr', 'data-cy-files-list-row-name').should('equal', `${randomFileName}.jpg`)
+			getRowForFileId(movFileId).should('have.length', 1).invoke('attr', 'data-cy-files-list-row-name').should('equal', `${randomFileName}.mov`)
+		})
+
+		it('Copies both files when copying the .jpg', () => {
+			copyFile(`${randomFileName}.jpg`, '.')
+			refreshView('All files')
+
+			getRowForFile(`${randomFileName}.jpg`).should('have.length', 1)
```
Comment (come-nc) [inline cypress/e2e/files/live_photos.cy.ts:128]: What about a `fileShouldExist` function? or `assertFileExists`.
And `assertFileNameById` for the name tests below.

**Review by come-nc — COMMENTED** [review]
:heart:

File: cypress/e2e/files/live_photos.cy.ts (hunk 1 above)
Comment (artonge) [reply cypress/e2e/files/live_photos.cy.ts:128]: Could make sense, but the current state is already readable enough IMO.
Can be done in a follow-up if needed.

**Review by skjnldsv — APPROVED** [review]

File: apps/files/lib/Listener/SyncLivePhotosListener.php [hunk 2]
```diff
@@ -52,37 +47,40 @@ class SyncLivePhotosListener implements IEventListener {
...
-			$peerFile = $this->getLivePhotoPeer($event->getSource()->getId());
+			$peerFileId = $this->livePhotosService->getLivePhotoPeerId($event->getSource()->getId());
+		}
+
+		if ($peerFileId === null) {
+			return; // Not a live photo.
 		}
 
+		// Check the user's folder.
+		$peerFile = $this->userFolder->getFirstNodeById($peerFileId);
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/files/lib/Listener/SyncLivePhotosListener.php:80]: True, but outside the scope of this PR.

**Pull request merged**